code/game/machinery/vending.dm

```diff
@@ -1320,3 +1316,128 @@
 	component_parts += new /obj/item/weapon/vending_refill/crittercare(0)
 	component_parts += new /obj/item/weapon/vending_refill/crittercare(0)
 	component_parts += new /obj/item/weapon/vending_refill/crittercare(0)
+
+//CODING BOOGALOO : FLIPPING OUT EDITION//
+//This is a WIP segment of code that will allow vending machines to be tipped over, and used as barricades///
+
+/*List of things to do:
+1.Add chance for projectiles to be blocked/pass
+2.Add checks to stop simple mobs from using these verbs
+3.Add checks to only show topple if it's not tipped and stand up if it's tipped
+4.Prevent anchoring while tipped
+5.Allow mobs to stand on the same tile as the machine when tipped, just like table barricades
+6.Let items pass over the top of vending machines like seriously what the fuck I thought I just needed to add the flag
+7.Learn to code instead of just scabbing code from other .dm files and just changing it around
+
+List of things I need to run past Frosty first but would like to implement:
+1.Hacked machines have a chance to change the price charged ranging from free to 5x original price
+2.Dragging/Pushing vending machines slows you down
+3.Custom vending machines for cargo to load items into to sell
+*/
+
+/obj/machinery/vending/verb/tip(mob/user in view(1))
+	set name = "Tip over"
+	set category = "Object"
+	set src in view(1)
+	if(tippable = 0)
+		to_chat(user, "<span class='notice'>How do you propose to tip [src]?</span>")
+		return
+	if(anchored != 0)
+		to_chat(user, "<span class='notice'>[src] is firmly bolted to the floor</span>")
+		return
+	if(tipped != 0)
+		to_chat(user, "<span class='notice'>[src] is already tipped over.</span>")
+		return
+	else
+		src.visible_message("<b><font color=red>[user] starts to rock [src] back and forth!</font></b>")
+		do_after(user, 80, target = src)
+		src.icon_state = "liberationstation"
+		src.visible_message("<b><font color=red>[user] tips [src] to the ground with a thud!</font></b>")
+		climbable = !climbable
+		tipped = !tipped
+		pass_flags = LETPASSTHROW
+		return
+
+/obj/machinery/vending/verb/untip(mob/user in view(1))
+	set name = "Stand up"
+	set category = "Object"
+	set src in view(1)
+	src.visible_message("<b><font color=red>[user] starts to stand [src] back up!</font></b>")
+	do_after(user, 120, target = src)
+	src.icon_state = "[initial(icon_state)]"
+	src.visible_message("<b><font color=red>[user] stands [src] upright!</font></b>")
+	tipped = !tipped

```

Comment:
Again here, the function of this proc is to right the vending machine. If you've reached this stage then you know the vending machine should be righted. So set it to `tipped = FALSE` or `tipped = 0` whatever you're going with here.